contracts/modules/Checkpoint/EtherDividendCheckpoint.sol

```diff
@@ -164,15 +100,18 @@ contract EtherDividendCheckpoint is ICheckpoint, Module {
      * @param _dividendIndex Dividend to pay
      */
     function _payDividend(address _payee, Dividend storage _dividend, uint256 _dividendIndex) internal {
-        uint256 claim = calculateDividend(_dividendIndex, _payee);
+        (uint256 claim, uint256 withheld) = calculateDividend(_dividendIndex, _payee);
         _dividend.claimed[_payee] = true;
         _dividend.claimedAmount = claim.add(_dividend.claimedAmount);
-        if (claim > 0) {
-            if (_payee.send(claim)) {
-              emit EtherDividendClaimed(_payee, _dividendIndex, claim);
+        uint256 claimAfterWithheld = claim.sub(withheld);
+        if (claimAfterWithheld > 0) {
+            if (_payee.send(claimAfterWithheld)) {
```

Comment:
The use-case we were trying to address is when sending multiple payments, not failing the whole batch if one of the payments failed.

The `send` is wrapped in an if statement which should manually revert corresponding changes (and emit a `EtherDividendClaimFailed` event) if the send fails.

There was a bug which I've just fixed as well and I added a test case for this as well.

@satyamakgec Can you think of any other security issues with this approach?

Reply:
understood the use case, I don't think so any more security issue with this approach as `send` and `transfer` works very similarly only difference is `send` return boolean and `transfer` revert so in case of send we need to reset all state variables if we get `false` (You already did that). So I think we are fine with this approach the only issue was the wrong event emission that you fixed.